src/main/java/vn/triplet/controller/admin/UserController.java

```diff
@@ -1,14 +1,74 @@
 package vn.triplet.controller.admin;
 
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import vn.triplet.model.User;
+import vn.triplet.model.User.Role;
+import vn.triplet.service.UserService;
 
 @Controller
 public class UserController {
 
+	private static final Logger logger = Logger.getLogger(UserController.class);
+	@Autowired
+	UserService userService;
+
 	@RequestMapping("/admin/users")
-	public String index() {
+	public String index(Model model) {
+		List<User> users = userService.loadUsers();
+		model.addAttribute("users", users);
 		return "views/admin/user/users";
 	}
 
+	@RequestMapping("/admin/users/{id}")
+	public String show(@PathVariable("id") int id, Model model) {
+		User user = userService.findById(id);
+		model.addAttribute("user", user);
+		return "views/admin/user/user";
+	}
+
+	@RequestMapping(value = "admin/users/{id}/delete", method = RequestMethod.GET)
+	public String deleteUser(@PathVariable("id") Integer id, final RedirectAttributes redirectAttributes) {
+		logger.info("delete user");
+		userService.deleteUser(id);
+		return "redirect:/admin/users";
+
+	}
+
+	@RequestMapping(value = "admin/users/add", method = RequestMethod.GET)
+	public String newStudent(Model model) {
+		logger.info("new user");
+		User user = new User();
+		// set default value
+		user.setRole(Role.USER);
+		model.addAttribute("userForm", user);
+		model.addAttribute("status", "add");
+		return "views/admin/user/user-form";
+	}
+
+	@RequestMapping(value = "/admin/users/saveupdate")
+	public String saveOrUpdate(@ModelAttribute("userForm") User user) {
+		logger.info(user.getName());
+		userService.saveOrUpdate(user);
+		return "redirect:/admin/users";
+	}
+	
+	@RequestMapping(value="admin/users/{id}/edit")
+	public String editUser(@PathVariable("id") int id, Model model ) {
+		User user = userService.findById(id);
```

Comment:
xử lý khi user ko tìm thấy nữa 
code này bị lặp nè, nên tạo 1 method private rồi gọi cho các action cần nhé

Reply:
Mỗi hàm em xử lý cái not found một kiểu chị